pkg/sql/colexec/external/external.go

```diff
@@ -650,6 +673,217 @@ func ScanFileData(param *ExternalParam, proc *process.Process) (*batch.Batch, er
 	return bat, nil
 }
 
+func getFileService(param *ExternalParam, dir string) (fileservice.FileService, error) {
+	c := fileservice.Config{
+		Name:    defines.LocalFileServiceName,
+		Backend: "DISK",
+		DataDir: dir,
+	}
+	if param.extern.ScanType == tree.S3 {
+		c.Name = defines.S3FileServiceName
+		c.Backend = "S3"
```

Comment:
This is wrong. The backend of 'S3' should be configured in the config file. The initialization of proc.FS already handles these details. Do use proc.FS instead.